publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -179,6 +179,13 @@ public Publication createPublicationFromImportedEntry(Publication inputData, Imp
         return insertResource(newResource);
     }
 
+    public Publication updatePublicationByImportEntry(Publication publication, ImportSource importSource) {
+        if (nonNull(importSource)) {
+            publication.addImportDetail(new ImportDetail(Instant.now(), importSource));
+        }
```

Comment:
Throw exception if null instead? Doesn't really make sense to use this method if `importSource` is null

Reply:
throwing IllegalArgumentException when passing null to this method